qiskit/algorithms/gradients/base_estimator_gradient.py

```diff
@@ -0,0 +1,159 @@
+
+    def __init__(
+        self,
+        estimator: BaseEstimator,
+        **run_options,
+    ):
+        """
+        Args:
+            estimator: The estimator used to compute the gradients.
+            run_options: Backend runtime options used for circuit execution. The order of priority is:
+                run_options in `run` method > gradient's default run_options > primitive's default
+                setting. Higher priority setting overrides lower priority setting.
+        """
+        self._estimator: BaseEstimator = estimator
+        self._default_run_options = run_options
+
+    def run(
+        self,
+        circuits: Sequence[QuantumCircuit],
+        observables: Sequence[BaseOperator | PauliSumOp],
+        parameter_values: Sequence[Sequence[float]],
+        parameters: Sequence[Sequence[Parameter] | None] | None = None,
+        **run_options,
+    ) -> EstimatorGradientResult:
+        """Run the job of the estimator gradient on the given circuits.
+
+        Args:
+            circuits: The list of quantum circuits to compute the gradients.
+            observables: The list of observables.
+            parameter_values: The list of parameter values to be bound to the circuit.
+            parameters: The Sequence of Sequence of Parameters to calculate only the gradients of
+                the specified parameters. Each Sequence of Parameters corresponds to a circuit in
+                `circuits`. Defaults to None, which means that the gradients of all parameters in
+                each circuit are calculated.
+            run_options: Backend runtime options used for circuit execution. The order of priority is:
+                run_options in `run` method > gradient's default run_options > primitive's default
+                setting. Higher priority setting overrides lower priority setting.
+
+        Returns:
+            The job object of the gradients of the expectation values. The i-th result corresponds to
+            ``circuits[i]`` evaluated with parameters bound as ``parameter_values[i]``.
+
+        Raises:
+            QiskitError: Invalid arguments are given.
+        """
+        # Validate the arguments.
+        self._validate_arguments(circuits, observables, parameter_values, parameters)
+        # The priority of run option is as follows:
+        # run_options in `run` method > gradient's default run_options > primitive's default setting.
+        run_opts = copy(self._default_run_options)
+        run_opts.update(**run_options)
+
+        job = PrimitiveJob(
+            self._run, circuits, observables, parameter_values, parameters, **run_opts
+        )
+        job.submit()
+        return job
+
+    @abstractmethod
+    def _run(
+        self,
+        circuits: Sequence[QuantumCircuit],
+        observables: Sequence[BaseOperator | PauliSumOp],
+        parameter_values: Sequence[Sequence[float]],
+        parameters: Sequence[Sequence[Parameter]] | None = None,
```

Comment:
Do you mean this will fail? (Sorry, I don't quite get it) We still need all parameter values to calculate partial gradients.  
`gradient.run(circuit=[qc1, qc2], observables=[obs1, obs2], parameter_values=[values1, values2], parameters=[param_subset1, param_subset2])`